acquire/bearer.go

```diff
@@ -84,10 +86,17 @@ func NewRemoteBearerTokenAcquirer(options RemoteBearerTokenAcquirerOptions) (*Re
 	}, nil
 }
 
+// Acquire provides the cached token or, if it's near its expiry time, contacts
+// the server for a new token to cache.
 func (acquirer *RemoteBearerTokenAcquirer) Acquire() (string, error) {
+	acquirer.lock.RLock()
 	if time.Now().Add(acquirer.options.Buffer).Before(acquirer.authValueExpiration) {
+		defer acquirer.lock.RUnlock()
```

Comment:
It might be a good idea to lock in the values in a local variables before return i.e. to avoid contingency on the edges
```
authValue := acquirer.authValue
acquirer.lock.RUnlock()
return authValue
```
Inspired by https://github.com/xmidt-org/webpa-common/blob/master/device/registry.go#L162

We want to do that at the end of this function as well on line 134